### DISCUSSION TITLE
Can you provide alternative key names to avoid collisions?

### DISCUSSION THREAD

**Discussion by oliverfoggin** [SwiftGen/SwiftGen#897] [Q&A] [answered]
I have started work on a mature project and I'm adding swiftgen to our strings (it is already used for the assets).

However, some of the string keys that we use (managed externally and so I'm not easily able to change them) contain keys like...

"something_string_something"

or 

"something_float_something"

I am using the `_` as a separator and so I end up with enums like `public enum String` and `public enum Float` which then collide with the functions...

```
public static func template(_ p1: Int) -> String {
```

As it now thinks the `String` return type here is the enum `String` that is defined in this file.

Is it possible to provide alternatives for the key so that swiftgen will create it like... `public enum Strng` and `public enum Flt` instead?

Obviously, if I could rename the strings easily I would but they're not so straight forward to change now.

Thanks

**Comment by oliverfoggin** [comment]
In the end I created a custom template and added an explicit `Swift.` to the type declarations. This has solved the issue.

Possibly something to add as an option in the future? Maybe?

Thanks

**Comment by djbe** [comment, marked as answer]
That was exactly what I was going to suggest (custom template), although I'd have added a check if a key was a `Int` or `String` and modified that (generated) key somehow (to for example `Strng` like you mentioned). But your solution might be a bit quicker to do.

This isn't something we'll be adding to our built-in templates, we'd have to maintain and test those extra cases, for very limited added value (your specific case).
